client/db/bolt/upgrades.go

```diff
@@ -28,6 +28,10 @@ var upgrades = [...]upgradefunc{
 	v3Upgrade,
 	// v3 => v4 splits orders into active and archived.
 	v4Upgrade,
+	// v4 => v5 adds PrimaryCredentials with determinstic client seed, but the
+	// only thing we need to do during the DB upgrade is to update the
+	// db.AccountInfo to differentiate legacy vs. new-style key.
+	v5Upgrade,
```

Comment:
I'm not _totally_ hung up on the idea of "authenticate upgrades", but I do have concerns I'd like to sort out, hopefully with just docs.
The main thing I'm wondering is what if there is a `v6Upgrade` unauthenticated upgrade here administered by client/db automatically on DB open, and it wants to do something with the structure of the account bucket and its sub-buckets and values, but now what if the first authenticated core version **have not been run yet**.  The two concerns are: (1) would v6 upgrade work if the core version upgrade associated with v5 hadn't been run yet, and (2) if v6 did work but login the first authenticated core versoin upgrade is going to be assuming v5, not v6 where it really is.

Start stacking up more unauthenticated and authenticate upgrades, and I can see this getting fragile.

I'm wondering more if there's a conceivable way to have the client/db-administered upgrades have a crypter argument that to all upgrades, even the authenticated ones, are in a single upgrade chain. If no crypter, break the upgrade chain and resume it on login.

Reply:
Good point. I bet we can just move all DB upgrades to `Login` and pass the `Crypter`?

Reply:
Seems like a good bet. Seem like it's gonna work out?

Reply:
There are some nasty consequences of doing DB upgrades to login.  It means there can be no DB access prior to login, which means stuff like checking for active orders and advising the user to login immediately can't happen.  Maybe other things too.

Reply:
Yeah, upgrade in `Login` gets messy real quick. Hmm.

Reply:
I think I'm going to try to make it work.